packages/grafana-data/src/types/datasource.ts

```diff
@@ -25,6 +25,12 @@ export type DataSourceQueryType<DSType> = DSType extends DataSourceApi<infer TQu
 // Utility type to extract the options type TOptions from a class extending DataSourceApi<TQuery, TOptions>
 export type DataSourceOptionsType<DSType> = DSType extends DataSourceApi<any, infer TOptions> ? TOptions : never;
 
+type DatasourceChangeHandler<TQuery> = (
+  prevSettings: DataSourceInstanceSettings,
+  newSettings: DataSourceInstanceSettings,
```

Comment:
Nicer might be something like:
```suggestion
  ds: DataSourceApi<TQuery, TOptions>,
  oldSettings: DataSourceInstanceSettings,  // can be any type
```

Many datasources have helper functions on the instance (not the settings)


Actually.... why do we want the function on the plugin and not on the DataSourceAPI directly?